src/EPR.Calculator.API/Controllers/CalculatorController.cs

```diff
@@ -52,5 +55,33 @@ public IActionResult GetCalculatorRun()
             // TODO: Return the details of a particular run
             return new OkResult();
         }
+
+        [HttpGet]
+        [Route("calculatorRunByName/{name}")]
+        public IActionResult GetCalculatorRunByName([FromRoute] string name)
+        {
+            if (!ModelState.IsValid)
+            {
+                return StatusCode(StatusCodes.Status400BadRequest, ModelState.Values.SelectMany(x => x.Errors));
+            }
+
+            var calculatorRun = _context.CalculatorRuns.Where(run => run.Name.ToLower() == name.ToLower());
+
+            if (calculatorRun.IsNullOrEmpty())
+            {
+                return new ObjectResult("No data found for this calculator name") { StatusCode = StatusCodes.Status404NotFound };
+            }
+
+            try
+            {
+                var currentRunDetails = CalculatorRunMapper.Map(calculatorRun.First());
```

Comment:
Based on my previous comment we can use this 
` CalculatorRunMapper.Map(calculatorRun);`
It will be just one record for the name